src/containers/FilterContainer.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Created by Katarzyna_Bak on 30.05.2017.
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { filterTask, searchTask } from '../actions';
+
+let FilterContainer = ({ dispatch }) => {
+
+    const handleChange = (e) => {
+        if(e.target.checked) {
+            dispatch(filterTask('SHOW_DONE'));
+        } else {
+            dispatch(filterTask('SHOW_ALL'));
+        }
+    };
+
+    const handleInputChange = (e) => {
+        dispatch(searchTask(e.target.value));
```

Comment:
Think of debouncing search, so it's not called on each key press. It shouldn't be a problem now, as all data is kept in memory. But if `searchTask` would trigger server request it can cause performance issues.